ocaml/lambda/translprim.ml

```diff
@@ -1064,8 +1064,22 @@ let specialize_primitive env loc ty ~has_constant_constructor prim =
         Some (Primitive (Pmakeblock(tag, mut, Some shape, mode),arity))
       else None
     end
+  | Primitive (Preuseblock { tag; mut; shape = None; resets; mode }, arity), fields ->
```

Comment:
It's not clear to me that this little optimization buys anything for memory re-use. @mshinwell ?